model-optimizer/mo/front_ng/extractor.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+import re
+from collections import defaultdict
+from copy import copy
+
+import numpy as np
+
+from mo.utils.error import Error
+
+
+def fe_decodeNameWithPort(inputModel, node_name: str):
```

Comment:
I don't see any unit test for this utility function. Please, add

Reply:
As discussed, added 'mock' frontend for unit testing. Appropriate tests were also added